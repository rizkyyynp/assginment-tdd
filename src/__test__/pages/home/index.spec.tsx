import React from "react";
import { fireEvent, render } from "@testing-library/react";
import HomePageView from "@/components/pages/home";

describe("render home", () => {
    it("should render home", () => {
        const { container } = render(<HomePageView />);
        expect(container).toMatchSnapshot();
    });
    it("should click button submit on home page", () => {
        const { getByText } = render(<HomePageView />);
        const buttonNext = getByText("Next");
        fireEvent.click(buttonNext);
        expect(buttonNext).toBeInTheDocument();

        const buttonPrevious = getByText("Previous");
        fireEvent.click(buttonPrevious);
        expect(buttonPrevious).toBeInTheDocument();

        const buttonReadmore = getByText("Read more");
        fireEvent.click(buttonReadmore);
        expect(buttonReadmore).toBeInTheDocument();
    });
});
