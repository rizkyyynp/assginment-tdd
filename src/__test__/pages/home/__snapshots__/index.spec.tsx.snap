// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render home should render home 1`] = `
<div>
  <div
    class="bg-abstract min-h-screen"
  >
    <div
      class="py-3.75 px-9% pb-25 home1:p-5"
    >
      <h1
        class="text-center pb-3.75 text-green2 text-5xl ml-12 md:ml-0 mb-10"
      >
        List of Users
      </h1>
      <div
        class="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-3.75 ml-10 px-5 md:ml-0 md:px-0"
      />
      <div
        class="flex justify-between w-full mt-5"
      >
        <button
          class="px-4 py-2 ml-12 md:ml-0 bg-green2 text-white rounded disabled:opacity-50"
          disabled=""
          type="button"
        >
          Previous
        </button>
        <p
          class="text-green2 font-bold"
        >
          1
        </p>
        <button
          class="px-4 py-2 bg-green2 text-white rounded disabled:opacity-50"
          type="button"
        >
          Next
        </button>
      </div>
    </div>
  </div>
</div>
`;
